build_tools/docker/context/setup_convperf.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Sets up `convperf` (https://github.com/nod-ai/convperf).
+#
+# `convperf` benchmarks convolution workloads on IREE and other backends such
+# as libxsmm. IREE is included as a submodule.
+#
+# Usage:
+#    ./setup_convperf.sh \
+#        <convperf repo dir> \
+#        <convperf sha>
+
+set -xeuo pipefail
+
+export REPO_DIR=$1
+export REPO_SHA=$2
+
+pushd ${REPO_DIR}
+# We use a fork of ConvPerf because the main branch relies on an outdated fork
+# of IREE and using an up-to-date version requires code changes.
+git clone --jobs 8 --depth 1 --recurse-submodules https://github.com/mariecwhite/convperf.git
+pushd convperf
+
+# Checkout a specific commit.
+git checkout ${REPO_SHA}
+
+# Create virtual environment.
+python3 -m venv convperf.venv
+source convperf.venv/bin/activate
+pip install -r requirements.txt
+
+popd # convperf
+
+# Since the root user clones the convperf repo, we update permissions so that a
+# runner can access this repo.
+chmod -R 777 .
+
+# Make sure there are no local changes to the IREE submodule since the workflow
+# updates this at each run.
```

Comment:
This is really a part of the chmod above. You just made the local changes, that's why you need to undo them. Together these are basically a hack for "make everything have maximal permissions except don't set the executable bit because git tracks that"

Reply:
Gotcha. Is there an action item here or just an FYI?

Reply:
I would just rearrange the comment a bit. I think also it makes sense to do the restore for not just the IREE repo. Left a suggested edit.